lib/opencensus/trace/integrations/faraday_middleware.rb

```diff
@@ -17,53 +17,87 @@
 module OpenCensus
   module Trace
     module Integrations
+      ##
+      # A middleware for Faraday that creates spans for outgoing requests, and
+      # propagates the trace context appropriately.
+      #
       class FaradayMiddleware < Faraday::Middleware
-        ##
-        # Create a Trace span with the HTTP request/response information.
-        def call env
-          OpenCensus::Trace.in_span "faraday_request" do |span|
-            add_request_labels span, env if span
-
-            response = @app.call env
+        ## The default name for Faraday spans
+        DEFAULT_SPAN_NAME = "faraday_request".freeze
 
-            add_response_labels span, env if span
+        ##
+        # Create a FaradayMiddleware.
+        #
+        # @param [#call] app Next item on the middleware stack.
+        # @param [SpanContext] span_context The span context within which
```

Comment:
If we're letting a user provide an `SpanContext` instance at initialization, we might want to let them specify on a request time via the request context.

Reply:
Agreed in principle, but I'm not sure what that looks like in Faraday. The [request class](https://github.com/lostisland/faraday/blob/master/lib/faraday/request.rb) doesn't seem to be extensible.

Reply:
I was looking at the last paragraph of https://github.com/lostisland/faraday#basic-use

Reply:
Ick. But I guess it works. Added support for it.